### PR TITLE
Screen HIL: Fix set_brightness

### PR DESCRIPTION
**Pull request by dcz-self**
The set_brightness method was not causing a callback, confusing the syscall driver and revealing a design/documentation mistake.

### Pull Request Overview

This pull request fixes an oversight in Display HIL.

I put some non-functional improvements to LPM013M126 driver on top.

### Testing Strategy

This pull request was tested by applying to a privte branch and running on LPM013M126 display.

### TODO or Help Wanted

This pull request still needs review.

### Documentation Updated

- [x] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [x] Ran `make prepush`.

### CODE REVIEW

**Review by alistair23 — APPROVED** [review]

**Comment by bradjc** [comment]
Isn't the fix to return an error instead of `Ok(())`? It looks like this isn't actually setting brightness.

**Comment by dcz-self** [comment]
Good point. The problem I aimed to fix here was "hangs instead of progressing". I didn't consider what the return value for "I don't do brightness" should be. NOSUPPORT perhaps?

**Comment by bradjc** [comment]
> Good point. The problem I aimed to fix here was "hangs instead of progressing". I didn't consider what the return value for "I don't do brightness" should be. NOSUPPORT perhaps?

agreed

**Comment by dcz-self** [comment]
I have no idea why it started failing. I even rebased. This doesn't seem related at first glance.

**Comment by brghena** [comment]
It's not your fault that it's failing. There's a CI bug right now causing all PRs to fail and we're working on it. You can ignore the `litex-sim-ci` for now.

File: capsules/src/lpm013m126.rs [hunk 1]
```diff
@@ -543,7 +543,17 @@ where
 
     fn set_brightness(&self, _brightness: usize) -> Result<(), ErrorCode> {
         // TODO: add LED PWM
-        Ok(())
+        let scheduled = schedule_deferred(
+            self.deferred_caller,
+            &self.command_complete_callback,
+            "set_brightness complete",
+        );
```
Comment (bradjc) [inline capsules/src/lpm013m126.rs:550]: Wait why does this need to set a deferred call? The client should not expect a callback if the `set_brightness()` command fails.

Comment (dcz-self) [reply capsules/src/lpm013m126.rs]: You're right, I did that change too mechanically.

**Review by bradjc — APPROVED** [review]

**Review by alistair23 — APPROVED** [review]

**Pull request merged**